mlflow/store/tracking/file_store.py

```diff
@@ -931,3 +970,81 @@ def record_logged_model(self, run_id, mlflow_model):
             self._set_run_tag(run_info, tag)
         except Exception as e:
             raise MlflowException(e, INTERNAL_ERROR)
+
+
+def _like_pattern_to_regex(pattern):
+    """
+    Convert a pattern string for LIKE and ILIKE to the equivalent regular expression.
+    """
+    return "^" + pattern.replace("%", ".*") + "$"
+
+
+def _comparator_to_func(comparator, value):
+    if comparator == "=":
+        return value.__eq__
+    elif comparator == "!=":
+        return value.__ne__
+    elif comparator in ("LIKE", "ILIKE"):
+        flag = re.IGNORECASE if comparator == "ILIKE" else 0
+        regex = re.compile(_like_pattern_to_regex(value), flag)
+        return regex.match
+    raise MlflowException.invalid_parameter_value(f"Invalid comparator for attribute: {comparator}")
+
+
+def filter_by_attribute(key, comparator, value):
+    f = _comparator_to_func(comparator, value)
+    return lambda exp: f(getattr(exp, key))
+
+
+def filter_by_tags(key, comparator, value):
+    f = _comparator_to_func(comparator, value)
+    return lambda exp: any(f(tag[1]) for tag in filter(lambda tag: tag[0] == key, exp.tags.items()))
+
+
+def _get_search_experiments_filters(parsed_filters):
+    filters = []
+    for f in parsed_filters:
+        type_ = f["type"]
+        key = f["key"]
+        comparator = f["comparator"]
+        value = f["value"]
+        if type_ == "attribute":
+            f = _comparator_to_func(comparator, value)
+            filters.append(filter_by_attribute(key, comparator, value))
+        elif type_ == "tag":
+            f = _comparator_to_func(comparator, value)
+            filters.append(filter_by_tags(key, comparator, value))
+        else:
+            raise MlflowException.invalid_parameter_value(f"Invalid token type: {type_}")
+    return filters
+
+
+# https://stackoverflow.com/a/56842689
+class _Reversor:
```

Comment:
Do we need to also implement `__le__`, `__gt__`, `__ge__` method ?

Reply:
We don't: https://stackoverflow.com/questions/37693373/how-to-sort-a-list-with-two-keys-but-one-in-reverse-order/56842689#comment105013524_56842689

Reply:
Nice. We can add the link in doc: https://docs.python.org/3/library/functools.html#functools.total_ordering

Reply:
> Total ordering is unnecessary for use as a key parameter in the sorted function. You only need == and <